pkg/kubelet/watchdog/watchdog_linux.go

```diff
@@ -166,10 +166,32 @@ func (hc *healthChecker) Start(ctx context.Context) {
 }
 
 func (hc *healthChecker) doCheck() error {
-	for _, hc := range hc.getHealthCheckers() {
-		if err := hc.Check(nil); err != nil {
-			return fmt.Errorf("checker %s failed: %w", hc.Name(), err)
+	logger := klog.Background()
+	var uninitializedCheckers []string
+
+	for _, checker := range hc.getHealthCheckers() {
+		// Check if the health checker has completed initialization.
+		// If it implements InitializableHealthChecker and returns false,
+		// skip the check to avoid false alarms during startup.
+		if !healthz.IsHealthCheckerInitialized(checker) {
+			uninitializedCheckers = append(uninitializedCheckers, checker.Name())
+			logger.V(4).Info("Health checker not yet initialized, skipping check",
+				"checker", checker.Name())
+			continue
+		}
```

Comment:
this is functionally equivalent to @SergeyKanzhelev approach. Ultimately, we assume that the components which expose a health check _but are not started yet_ are healthy, instead of the previous state on which we assumed unhealthy. There's AFAIK no way to expose a tri-state healthy/unhealthy/unknown (or "not yet") and to move this decision to the caller. Or there is?

Reply:
> Failure to fully initialize kubelet should probably be a separate health check with it's own timeout.

I don't have a good idea for this approach yet. If there's a clear direction, I'd be very willing to work on it.

Reply:
I'd be generally favor of splitting startup vs steady state health check, but I'm not sure we have infrastructure for that (e.g. 2 separate and acknowledged endpoints)